### ISSUE TITLE
Leave rendered HTML behind on deactivation

### ISSUE THREAD

**Issue by brashrebel** [brashrebel/render#72]
We wouldn't be the first to do this and I think it is a cool practice. If the plugin gets deactivated, can we make all the shortcodes that were in use leave behind their HTML output instead of the `[shortcode]`? Not sure if this is a good idea for every shortcode but for many it would be nice and appreciated. @shackep has been requesting it.

**Comment by joelworsham** [comment]
Couldn't this be potentially devastating to those that want to continue using and tweaking the shortcodes? They would have to manually go back through and turn them into shortcodes again.

**Comment by joelworsham** [comment]
Styling of the HTML would be very hard, if not impossible as well, without a total revamp of the shortcode styling, because it's all currently done through CSS

**Comment by brashrebel** [comment]
Your first point is solid. They would potentially unless we had some crazy way of magically turning them back into shortcodes. As to your second point, I think we'd have to accept some loss of styling but it would be much better than just displaying [shortcode]. A button might just turn into a regular hyperlink for example which isn't terrible if you ask me.

And things like that wouldn't be a big problem when turning the plugin back on. The styling would work again...maybe...although, no it probably wouldn't because the shortcode triggers the enqueueing....hmmm....

I don't know how to solve this issue. It may be that SOME shortcodes could be candidates for converting on deactivation while others just create too many problems.

**Comment by joelworsham** [comment]
What's the benefit to doing this as a whole? Wouldn't them going back to "pre-Render" be incentive to keep the plugin? Just trying to wrap my brain around it.

**Comment by shackep** [comment]
I think Kyle wants to see if it can be done. :) I only showed him that Nick Haskins was building this feature it into aesop story engine. I think if you aren't the one who created the short codes you shouldn't be the ones to hard set them. But... if you wanted to create an add-on that would "make it so" it could be a handy tool to reduce plugins and hard bake a site that was build using shortcodes and a so so plugin..

**Comment by brashrebel** [comment]
Right. I think we'll start on case by case basis and begin evaluating the most important Render shortcodes and explore a deactivation/reactivation script.

Regarding shortcodes not made by Render, I don't want us to do anything that could interfere with them or imply that somehow we support them unless we specifically create an integration for that plugin or theme. A deactivation/reactivation script would need to only act on shortcodes which we create as otherwise we'd likely have a support nightmare.

**Comment by brashrebel** [comment]
This is really important I think for our next version. @bearded-avenger wrote this on his blog: http://nickhaskins.com/2015/01/brackets-aint-empty-cuz/

**Comment by joelworsham** [comment]
I think that having the checkbox option on uninstall is the way to go (with it set to off by default)
